File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/SearchTextBox/Features/SearchTextBoxIsSearchActivePage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.SearchTextBox.Features
{
    public partial class SearchTextBoxIsSearchActivePage : UserControl
    {
        public SearchTextBoxIsSearchActivePage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles Click event for Button control
        /// Sets tooltip text and icon according to search state
        /// </summary>
        private void btnCheck_Click(object sender, EventArgs e)
        {
            
            if (searchTextBox1.IsSearchActive)
            {
                this.errorProvider1.Icon = new Gizmox.WebGUI.Common.Resources.ImageResourceHandle("16x16.InProgress.png");
                errorProvider1.SetError(searchTextBox1, "Search in progress");
            }
            else
            {
                this.errorProvider1.Icon = new Gizmox.WebGUI.Common.Resources.ImageResourceHandle("16x16.Success.png");
                errorProvider1.SetError(searchTextBox1, "Search complete");
            }
        }

        /// <summary>
        /// Handles Search event for SearchTextBox control
        /// Hides tooltip for SearchTextBox control
        /// </summary>
        private void searchTextBox1_Search(object sender, EventArgs e)
        {
            this.errorProvider1.Icon = null;
            errorProvider1.SetError(searchTextBox1, "");
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/SearchTextBox/Features/SearchTextBoxIsSearchActivePage.vb
Namespace CompanionKit.Controls.SearchTextBox.Features

	Public Class SearchTextBoxIsSearchActivePage

        Public Sub New()
            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()
        End Sub

        ''' <summary>
        ''' Handles Click event for Button control
        ''' Sets tooltip text and icon according to search state
        ''' </summary>
        Private Sub btnCheck_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCheck.Click
            If Me.searchTextBox1.IsSearchActive Then
                Me.errorProvider1.Icon = New ImageResourceHandle("16x16.InProgress.png")
                Me.errorProvider1.SetError(Me.searchTextBox1, "Search in progress")
            Else
                Me.errorProvider1.Icon = New ImageResourceHandle("16x16.Success.png")
                Me.errorProvider1.SetError(Me.searchTextBox1, "Search complete")
            End If
        End Sub

        ''' <summary>
        ''' Handles Search event for SearchTextBox control
        ''' Hides tooltip for SearchTextBox control
        ''' </summary>
        Private Sub searchTextBox1_Search(ByVal sender As Object, ByVal e As EventArgs) Handles searchTextBox1.Search
            Me.errorProvider1.Icon = Nothing
            Me.errorProvider1.SetError(Me.searchTextBox1, "")
        End Sub

	End Class

End Namespace